lib/webmock/http_lib_adapters/net_http.rb

```diff
@@ -72,78 +72,57 @@ def constants(inherit=true)
         end
 
         def request(request, body = nil, &block)
+          return super unless started?
+
           request_signature = WebMock::NetHTTPUtility.request_signature_from_request(self, request, body)
 
           WebMock::RequestRegistry.instance.requested_signatures.put(request_signature)
 
           if webmock_response = WebMock::StubRegistry.instance.response_for_request(request_signature)
-            @socket = Net::HTTP.socket_type.new
```

Comment:
There's no reason to replace the socket here. At the very least, we'd want to close the old socket before initializing a new one.